x-pack/legacy/plugins/maps/public/layers/sources/es_agg_source.js

```diff
@@ -46,34 +46,8 @@ export class AbstractESAggSource extends AbstractESSource {
     }) : [];
   }
 
-  createField({ fieldName, label }) {
-
-    //if there is a corresponding field with a custom label, use that one.
-    if (!label) {
-      const matchField = this._metricFields.find(field => field.getName() === fieldName);
-      if (matchField) {
-        label = matchField.getLabel();
-      }
-    }
-
-    if (fieldName === COUNT_PROP_NAME) {
-      return new ESAggMetricField({
-        aggType: COUNT_AGG_TYPE,
-        label: label,
-        source: this,
-        origin: this.getOriginForField()
-      });
-    }
-    //this only works because aggType is a fixed set and does not include the `_of_` string
-    const [aggType, docField] = fieldName.split(AGG_DELIMITER);
-    const esDocField = new ESDocField({ fieldName: docField, source: this });
-    return new ESAggMetricField({
-      label: label,
-      esDocField,
-      aggType,
-      source: this,
-      origin: this.getOriginForField()
-    });
+  createField({ fieldName }) {
```

Comment:
Great simplification! The main consequence now is that `EsAggSource#createField` is no longer a factory-function and creates inconsistencies with the other implementations of `VectorSource#createField`.

1) By using the  `getMetricFieldForName`, which only will return previously created objects, this continues to work as expected because all object-instances are guaranteed to be read-only, so reuse is not an issue. But that's implicit and never enforced (ie. redux requires persistence to always flow through the store, and we enforce this through code-review).

2) The other implication is that `ESAggSource#createField` can _only_ construct fields that are actually available in the  `EsAggSource`, which is inconsistent with all the other implementations.  It is a fine limitation, but implementations should be consistent. e.g. consider EMS-sources disallowing creating fields which are not present in the manifest, ES-source disallowing creating fields which are not present in the index-pattern, etc...).

This was the reason we weren't reusing `getMetricFieldForName` here originally, instead doing a manual lookup for a custom label first.

3) `createField` can no longer be used transparently, regardless of caller. e.g. sometimes it is used in the Source#constructor as a shorthand (which would require creation-of-fields from scratch). Other times, it's used from outside (like here, where the bug arises https://github.com/elastic/kibana/blob/05f765a7a948bbd43ce3dad630cdaa148f888a74/x-pack/legacy/plugins/maps/public/layers/styles/vector/vector_style.js#L455-L457). 


Maybe consider (?)

- Introduce something like `VectorSource#getFieldByName`. `EsAggSource` can implement it just with `getMetricFieldByName`, the other sources can just delegate to use `createField` by default.
- Modify https://github.com/elastic/kibana/blob/05f765a7a948bbd43ce3dad630cdaa148f888a74/x-pack/legacy/plugins/maps/public/layers/styles/vector/vector_style.js#L455-L457 to `return this._source.getFieldByName(descriptor.name)`